sklearn/ensemble/_weight_boosting.py

```diff
@@ -1165,10 +1174,39 @@ def _boost(self, iboost, X, y, sample_weight, random_state):
         # Boost weight using AdaBoost.R2 alg
         estimator_weight = self.learning_rate * np.log(1.0 / beta)
 
-        if not iboost == self.n_estimators - 1:
+        if iboost == 0:
+            update_weight_method = "update"
+        elif iboost == self.n_estimators - 1:
+            update_weight_method = "do_nothing"
+        else:
+            if self.estimator_errors_[iboost - 1] > estimator_error:
+                if self.no_improvement == "reset_weights":
+                    update_weight_method = "reset"
+                elif self.no_improvement == "stop":
+                    warnings.warn(
+                        "Terminates training because the estimator error has increased."
+                    )
```

Comment:
No need to warn in this case. The user explicitly asked to stop in this case.

Reply:
I removed it.